File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/part-two-php-starter/src/Catalog/Service.php
<?php

namespace App\Catalog;

class Service
{
    public function __construct(
        public readonly string $name,
        public readonly string $description,
        public readonly bool $hasHttp,
        public readonly bool $hasGrpc,
        public readonly string $repositoryUrl,
    )
    {
    }

    public static function fromArray(array $data): static
    {
        return new static(
            name: $data['name'],
            description: $data['description'],
            hasHttp: $data['has_http'],
            hasGrpc: $data['has_grpc'],
            repositoryUrl: $data['github'],
        );
    }
}

File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/part-two-java-starter/src/main/java/lab/catalog/Service.java
package lab.catalog;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Service {
    public String name;
    public String description;
    @JsonProperty("has_http")
    public boolean hasHttp;
    @JsonProperty("has_grpc")
    public boolean hasGrpc;
    @JsonProperty("github")
    public String repositoryUrl;
}
